ui/plugin-system/src/runtime/DataQueriesProvider/DataQueriesProvider.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createContext, useCallback, useContext, useMemo } from 'react';
```

Comment:
Decided to not use `use-context-selector` here because all context values are updated every time the query changes so even if we use selector, all components will be triggered to update anyways

Reply:
I think we may benefit from some perf tuning around context, but that'll be better to do as a follow-up in the future. Agreed with not worrying about it in this particular PR.

Reply:
Agreed, I was talking to steven and mentioned that TimeRangeProvider might be a good candidate for `use-context-selector` because not all values are updated at the same time. For instance, absolute time range doesn't always change even when you refresh the time range